drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/conversion/GuidedDecisionTableGeneratorListener.java

```diff
@@ -115,80 +118,119 @@ public CaseInsensitiveMap getProperties() {
     }
 
     public Package getRuleSet() {
-        throw new UnsupportedOperationException( "Use getGuidedDecisionTables() instead." );
+        throw new UnsupportedOperationException("Use getGuidedDecisionTables() instead.");
     }
 
     public List<Import> getImports() {
-        return RuleSheetParserUtil.getImportList( getProperties().getProperty( DefaultRuleSheetListener.IMPORT_TAG ) );
+        return RuleSheetParserUtil.getImportList(getProperties().getProperty(DefaultRuleSheetListener.IMPORT_TAG));
     }
 
     public List<Global> getGlobals() {
-        return RuleSheetParserUtil.getVariableList( getProperties().getProperty( DefaultRuleSheetListener.VARIABLES_TAG ) );
+        return RuleSheetParserUtil.getVariableList(getProperties().getProperty(DefaultRuleSheetListener.VARIABLES_TAG));
     }
 
     public List<String> getFunctions() {
-        return getProperties().getProperty( DefaultRuleSheetListener.FUNCTIONS_TAG );
+        return getProperties().getProperty(DefaultRuleSheetListener.FUNCTIONS_TAG);
     }
 
     public List<String> getQueries() {
-        return getProperties().getProperty( DefaultRuleSheetListener.QUERIES_TAG );
+        return getProperties().getProperty(DefaultRuleSheetListener.QUERIES_TAG);
     }
 
     public List<String> getTypeDeclarations() {
-        return getProperties().getProperty( DefaultRuleSheetListener.DECLARES_TAG );
+        return getProperties().getProperty(DefaultRuleSheetListener.DECLARES_TAG);
     }
 
     public List<GuidedDecisionTable52> getGuidedDecisionTables() {
         return _dtables;
     }
 
-    public void startSheet( final String name ) {
+    public void startSheet(final String name) {
     }
 
     public void finishSheet() {
         this._propertiesListener.finishSheet();
         finishRuleTable();
     }
 
-    public void newRow( final int rowNumber,
-                        final int columns ) {
-        if ( rowNumber > this._ruleStartRow + LABEL_ROW ) {
+    public void newRow(final int rowNumber,
+                       final int columns) {
+        if (rowNumber > this._ruleStartRow + LABEL_ROW) {
             this._isNewDataRow = true;
         }
+        assertValueBuildersRowData();
     }
 
-    public void newCell( final int row,
-                         final int column,
-                         final String value,
-                         int mergedColStart ) {
+    private void assertValueBuildersRowData() {
```

Comment:
This ensures all "Source Builders" have the expected number of row data elements when a new row is processed (otherwise ```ExcelParser``` skips ```null``` cells from POI and "Source Builders" may contain less - and even worse - incorrectly ordered - row data).